compiler/rustc_codegen_cranelift/src/base.rs

```diff
@@ -879,8 +879,7 @@ fn codegen_array_len<'tcx>(
     match *place.layout().ty.kind() {
         ty::Array(_elem_ty, len) => {
             let len = fx
-                .monomorphize(&len)
-                .eval(fx.tcx, ParamEnv::reveal_all())
```

Comment:
`eval(...).eval_usize(...)` shouldn't be necessary, so I removed the first eval here. cc @bjorn3 for the cg_clif changes

Reply:
Nothing breaks when I remove the `.eval()` call. All cg_clif changes LGTM.